quickstart doc (#259)
qdrant/landing_page: qdrant-landing/content/documentation/quick-start.md

aliases:
  - quick_start
---
# Quickstart

## Recommended Workflow

> There are two ways of using Qdrant locally: [1) Local Mode](#option-1-local-mode) and [2) Docker Mode](#option-2-docker-mode). If you are a Python developer, we recommend that you first try Local Mode in [Qdrant Client](https://github.com/qdrant/qdrant-client), as it only takes a few moments to get setup. Then, you may further experiment with Qdrant Docker containers. When you are more comfortable with Qdrant, then you should deploy your app to a Free Tier [Qdrant Cloud](../cloud/quickstart-cloud/) cluster.

Reviewer: Again, I don't like Local Mode as first. We only really show it with Python . You can state in the docker instructions, or in the beginning, that if you will be working with Python and prototyping or debugging, then Local Mode may be simpler for you.

> There are two ways of using Qdrant locally: [1) Local Mode](#option-1-local-mode) and [2) Docker Mode](#option-2-docker-mode). We recommend that you first try Local Mode in [Qdrant Client](https://github.com/qdrant/qdrant-client), as it only takes a few moments to get setup. Then, you may further experiment with Qdrant Docker containers. When you are more comfortable with Qdrant, then you should deploy your app to a Free Tier [Qdrant Cloud](../cloud/quickstart-cloud/) cluster.

Reviewer: Please start with Docker mode. It is more general, while it seems like we only have instruction for Python in local mode. 

We should change the order down below as well.

Author: In what situations do we want people to use local mode?

> There are two ways of using Qdrant locally: [1) Local Mode](#option-1-local-mode) and [2) Docker Mode](#option-2-docker-mode). We recommend that you first try Local Mode in [Qdrant Client](https://github.com/qdrant/qdrant-client), as it only takes a few moments to get setup. Then, you may further experiment with Qdrant Docker containers. When you are more comfortable with Qdrant, then you should deploy your app to a Free Tier [Qdrant Cloud](../cloud/quickstart-cloud/) cluster.
> 

Reviewer: ```suggestion
There are two ways of using Qdrant locally: [1) Local Mode](#option-1-local-mode) and [2) Docker Mode](#option-2-docker-mode). If you are a Python developer, we recommend that you first try Local Mode in [Qdrant Client](https://github.com/qdrant/qdrant-client), as it only takes a few moments to get setup. Then, you may further experiment with Qdrant Docker containers. When you are more comfortable with Qdrant, then you should deploy your app to a Free Tier [Qdrant Cloud](../cloud/quickstart-cloud/) cluster.

```

client = QdrantClient(path="path/to/db")  # Persists changes to disk
```

## Option 2: Docker Mode

**Prerequisite:** First make sure Docker is installed and running on your system.

### Download the image

```bash
docker pull qdrant/qdrant
```

### Run the service

```bash
docker run -p 6333:6333 \

>     -v $(pwd)/qdrant_storage:/qdrant/storage \

Reviewer: ```suggestion
    -v $(pwd)/qdrant_storage:/qdrant/storage:z \
```

This will allow linux users with SELinux turned up to still use this and won't affect any users without.

    qdrant/qdrant
```

> Under the default configuration all data will be stored in the `./qdrant_storage` directory.

Reviewer: ```suggestion
Under the default configuration all data will be stored in the `./qdrant_storage` directory. This will also be the only directory that both the Container and the host machine can both see. 
```

Qdrant should now be accessible at [localhost:6333](http://localhost:6333)

### Initialize Qdrant Client 

```python
from qdrant_client import QdrantClient

client = QdrantClient("localhost", port=6333)
```

> <aside role="status">Qdrant has no encryption or authentication by default and new instances are open to everyone. Please read <a href="https://qdrant.tech/documentation/security/">Security</a> carefully for details on how to secure your instance.</aside>

Reviewer: ```suggestion
<aside role="status">By default, Qdrant starts with encryption or authentication . This means anyone with network access to your machine can access your Qdrant container instance. Please read <a href="https://qdrant.tech/documentation/security/">Security</a> carefully for details on how to secure your instance.</aside>
```

# Running vector search queries

> In this short example, you will create a Qdrant collection, load data into it and run a basic search query. 

Reviewer: ```suggestion
In this short example, you  will use the Python client to create a Qdrant collection, load data into it and run a basic search query. 
```

![Qdrant Quickstart](/docs/quickstart.png)

## Step 1. Create a collection

> You will be storing all of your vector data in a Qdrant collection. Let's call it `test_collection`. This collection will be using a doc production metric. 

Reviewer: ```suggestion
You will be storing all of your vector data in a Qdrant collection. Let's call it `test_collection`. This collection will be using a doc product distance metric to compare vectors. 
```

```python

from qdrant_client.http.models import Distance, VectorParams

client.recreate_collection(
    collection_name="test_collection",
    vectors_config=VectorParams(size=4, distance=Distance.DOT),
)
```

**Response:**

```python
True
```

> ## Step 2: Add vectors

Reviewer: I don't think it is a good idea to have differing payloads for each of the vectors. In the first vector, the city payload item is just a single entity while in the next 3 it is an array. 
And then the next two vectors do something else entirely and as a reader I am unsure what this means and why it is like this. 

For me trying to understand this code is harder than it need be just for a quick start. I do think it is important to show the flexibility of the payload but not here.

Author: How do you want the vectors to look? Can you write out the payloads?

Reviewer: ```
          {"id": 1, "vector": [0.05, 0.61, 0.76, 0.74], "payload": {"city": "Berlin" }},
          {"id": 2, "vector": [0.19, 0.81, 0.75, -0.11], "payload": {"city": "London" }},
          {"id": 3, "vector": [0.36, -0.55, 0.47, 0.94], "payload": {"city": "Moscow" }},
          {"id": 4, "vector": [0.18, 0.01, 0.85, 0.80], "payload": {"city": "New York" }}, 
          {"id": 5, "vector": [0.24, 0.18, 0.22, 0.44], "payload": {"city": "Beijing" }}, 
          {"id": 6, "vector": [0.35, 0.08, 0.11, 0.44], "payload": {"city": "Mumbai" }}, 
```

Reviewer: Sorry this makes you redo the sample queries and such below.

> Let's now add a few vectors with a payload:

Reviewer: ```suggestion
Let's now add a few vectors with a payload. Payloads are other data you want to associate with the vector:
```

        PointStruct(id=6, vector=[0.35, 0.08, 0.11, 0.44]),
    ]
)
print(operation_info)
```

**Response:**

```python
operation_id=0 status=<UpdateStatus.COMPLETED: 'completed'>
```

## Step 3: Run a query
Let's ask a basic question - Which of our stored vectors are most similar to the query vector `[0.2, 0.1, 0.9, 0.7]`?

```python
search_result = client.search(
    collection_name="test_collection",
    query_vector=[0.2, 0.1, 0.9, 0.7], 
    limit=3
)
print(search_result)
```

**Response:**

```python
ScoredPoint(id=4, version=0, score=1.362000031210482, payload={'city': ['London', 'Moscow']}, vector=None), 
ScoredPoint(id=1, version=0, score=1.2729999996721744, payload={'city': 'Berlin'}, vector=None), 
ScoredPoint(id=3, version=0, score=1.2080000013113021, payload={'city': ['Berlin', 'Moscow']}, vector=None)
```

> Note that payload and vector data is missing in these results by default.

Reviewer: ```suggestion
The results are returned in decreasing si,ilarity order. Note that payload and vector data is missing in these results by default.
```

I see the paylod in the response, just not the vector.
